src/Model/Model.ts

```diff
@@ -349,15 +347,18 @@ export class Model<T extends IModelData> implements IModel<T> {
    */
   public applyUpdates(changes: Partial<T> = {}, meta: Partial<IModelMeta<T>> = {}, relatedModels: any = {}): IModel<T> {
     relatedModels = { ...this.relatedModels, ...relatedModels };
+    let hasRelationship = false;
 
     if (!isEmpty(changes)) {
 
       // Validate the input, clear relatedModels whose ids may have just changed so they can be loaded again
       for (const key in changes) {
         this.checkFieldUpdateIsAllowed(key, changes[key]);
-        const relationship = find(this.relationships, { relatedFieldName: key });
-        if (relationship && relatedModels.hasOwnProperty(relationship.field)) {
-          delete relatedModels[relationship.field];
+        hasRelationship = some(this.relationships, { relatedFieldName: key });
```

Comment:
this will never be true because applyUpdates is called on the related model with the changes, in the case of a student and person relationship, the person.applyUpdates is called.  If I change the student.person.NameSuffix the apply updates will be called and we will check changes on the NameSuffix but it isn't a relationship on the person.